src/Writer.php

```diff
@@ -115,7 +115,7 @@ public function insertAll($records): int
     /**
      * Adds a single record to a CSV document
      *
-     * @param string[] $record an array
+     * @param (string|int|null)[] $record an array
```

Comment:
while I understand this notation I believe it is as incomplete as the one you are trying to fix for instance does your notation covers this usage ?

```php
$csv = Writer::createFromString('');
$csv->insertOne([
    new class {
        function __toString() {
            return 'foo';
        }
    },
    'bar',
]);

echo $csv->getContent();
```

Reply:
Hmm maybe it should just be `mixed[]` then?

Reply:
I'd rather use `array` and specify in the doc bloc that flat array consisting of scalar null and object which implements __toString values can be used IMHO mixed is just too permissive

Reply:
I don't think that is possible to express in the typehint though..

Reply:
no it's not that why I'm leaning toward just using the default typehint which is `array` and explain in the docblock exactly what is expected

Reply:
Right, I got ya.